src/sardana/macroserver/macros/standard.py

```diff
@@ -642,14 +642,17 @@ def run(self, motor, delta):
 ##########################################################################
 
 
-class ct(Macro):
+class ct(Macro, Hookable):
     """Count for the specified time on the active measurement group"""
 
     env = ('ActiveMntGrp',)
-
+    hints = {'allowsHooks': ('pre-acq', 'post-acq')}
     param_def = [
         ['integ_time', Type.Float, 1.0, 'Integration time']
     ]
+    # integration time has to be accessible from with in the hooks
+    # so declare it also as a property of the ct class
+    integ_time = 0
```

Comment:
I think there is no need to declare this on the class level. With setting the instance attribute `self.integ_time` in the run method it should be enough.

Reply:
okay, sure. I will change that in a bit.

Reply:
I could get rid of it, by accessing the integ_time via the measurementGroup, however, it seems that at the time of the pre_acq hook, the integ_time of the measurementGroup (getIntegrationTime()) is not updated to the current value of the current acquisition.

Reply:
Yes, for making it accessible already from the `pre-acq` hook it must be added to the instance. I think that's fine.